From d0667e6f3af8463ce7fdbb5b46071763db607890 Mon Sep 17 00:00:00 2001
From: Tomas Jelinek <tojeline@redhat.com>
Date: Fri, 22 Jan 2016 16:38:39 +0100
Subject: [PATCH] fix syntax error in utilization attributes functions

---
 pcsd/remote.rb | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pcsd/remote.rb b/pcsd/remote.rb
index 9805baf..484cce0 100644
--- a/pcsd/remote.rb
+++ b/pcsd/remote.rb
@@ -2277,7 +2277,7 @@ def set_resource_utilization(params, reqest, session)
 
   res_id = params[:resource_id]
   name = params[:name]
-  value = params[:value] if params[:value] else ''
+  value = params[:value] || ''
 
   _, stderr, retval = run_cmd(
     session, PCS, 'resource', 'utilization', res_id, "#{name}=#{value}"
@@ -2302,7 +2302,7 @@ def set_node_utilization(params, reqest, session)
 
   node = params[:node]
   name = params[:name]
-  value = params[:value] if params[:value] else ''
+  value = params[:value] || ''
 
   _, stderr, retval = run_cmd(
     session, PCS, 'node', 'utilization', node, "#{name}=#{value}"
-- 
1.8.3.1

