From 52f6b832f44455607ff4c735cfbcbfc2ec26e7c2 Mon Sep 17 00:00:00 2001
From: Tomas Jelinek <tojeline@redhat.com>
Date: Fri, 23 Oct 2015 16:40:06 +0200
Subject: [PATCH] fix cluster properties page in web UI

---
 pcsd/pcsd.rb | 47 +++++++++++++++++++++++------------------------
 1 file changed, 23 insertions(+), 24 deletions(-)

diff --git a/pcsd/pcsd.rb b/pcsd/pcsd.rb
index 83f3243..fccfd39 100644
--- a/pcsd/pcsd.rb
+++ b/pcsd/pcsd.rb
@@ -564,7 +564,7 @@ If checked, the cluster will refuse to start resources unless one or more STONIT
   allconfigoptions = []
   config_options.each { |i,k| k.each { |j| allconfigoptions << j } }
   ConfigOption.getDefaultValues(allconfigoptions)
-  ConfigOption.loadValues(allconfigoptions,cluster_name)
+  configOption_loadValues(allconfigoptions,cluster_name)
   return config_options
 end
 
@@ -623,6 +623,28 @@ class Node
   end
 end
 
+def configOption_loadValues(cos,cluster_name)
+  code,output = send_cluster_request_with_token(cluster_name, "get_cib")
+  $logger.info(code)
+  if code != 200
+    $logger.info "Error: unable to load cib"
+    $logger.info output
+    return
+  end
+
+  doc = REXML::Document.new(output)
+
+  cos.each {|co|
+    prop_found = false
+    doc.elements.each("cib/configuration/crm_config/cluster_property_set/nvpair[@name='#{co.configname}']") { |e|
+      co.value = e.attributes["value"]
+      prop_found = true
+    }
+    if prop_found == false
+      co.value = co.default
+    end
+  }
+end
 
 class ConfigOption
   attr_accessor :name, :configname, :type, :size, :units, :options, :default, :value, :desc
@@ -636,29 +658,6 @@ class ConfigOption
     @desc = desc
   end
 
-  def self.loadValues(cos,cluster_name)
-    code,output = send_cluster_request_with_token(cluster_name, "get_cib")
-    $logger.info(code)
-    if code != 200
-      $logger.info "Error: unable to load cib"
-      $logger.info output
-      return
-    end
-
-    doc = REXML::Document.new(output)
-
-    cos.each {|co|
-      prop_found = false
-      doc.elements.each("cib/configuration/crm_config/cluster_property_set/nvpair[@name='#{co.configname}']") { |e|
-      	co.value = e.attributes["value"]
-      	prop_found = true
-      }
-      if prop_found == false
-      	co.value = co.default
-      end
-    }
-  end
-
   def self.getDefaultValues(cos)
     metadata = `#{PENGINE} metadata`
     doc = REXML::Document.new(metadata)
-- 
1.9.1

