From 8321377072111f72855992bfc48fbc551022e7e5 Mon Sep 17 00:00:00 2001
From: Ivan Devat <idevat@redhat.com>
Date: Tue, 11 Oct 2016 12:50:36 +0200
Subject: [PATCH] disable gui

---
 pcsd/pcsd.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pcsd/pcsd.conf b/pcsd/pcsd.conf
index 0ac5eec..10904cf 100644
--- a/pcsd/pcsd.conf
+++ b/pcsd/pcsd.conf
@@ -3,7 +3,7 @@
 # Set PCSD_DEBUG to true for advanced pcsd debugging information
 PCSD_DEBUG=false
 # Set DISABLE_GUI to true to disable GUI frontend in pcsd
-PCSD_DISABLE_GUI=false
+PCSD_DISABLE_GUI=true
 # Set web UI sesions lifetime in seconds
 PCSD_SESSION_LIFETIME=3600
 # List of IP addresses pcsd should bind to delimited by ',' character
-- 
1.8.3.1

