--- pcs-0.9.139/pcsd/pcsd.rb.secure-cookie	2015-04-14 16:39:28.241833978 -0500
+++ pcs-0.9.139/pcsd/pcsd.rb	2015-04-14 16:40:12.639381902 -0500
@@ -31,7 +31,10 @@ end
 
 use Rack::Session::Cookie,
   :expire_after => 60 * 60,
-  :secret => secret
+  :secret => secret,
+  :secure => true, # only send over HTTPS
+  :httponly => true # don't provide to javascript
+
 
 #use Rack::SSL
 
@@ -45,8 +48,6 @@ also_reload 'pcs.rb'
 also_reload 'auth.rb'
 also_reload 'wizard.rb'
 
-enable :sessions
-
 before do
   $session = session
   $cookies = cookies
